addons/website_blog/tests/test_website_blog_flow.py

```diff
@@ -18,43 +31,66 @@ def test_website_blog_followers(self):
+            self.user_employee.partner_id, self.test_blog_post.message_partner_ids,
             'website_blog: subscribing to a blog should not subscribe to its posts')
         self.assertNotIn(
-            self.user_public.partner_id, test_blog_post.message_partner_ids,
+            self.user_public.partner_id, self.test_blog_post.message_partner_ids,
             'website_blog: subscribing to a blog should not subscribe to its posts')
 
         # Publish the blog
-        test_blog_post.write({'website_published': True})
+        self.test_blog_post.write({'website_published': True})
 
         # Check publish message has been sent to blog followers
-        publish_message = next((m for m in test_blog_post.blog_id.message_ids if m.subtype_id.id == self.ref('website_blog.mt_blog_blog_published')), None)
+        publish_message = next((m for m in self.test_blog_post.blog_id.message_ids if m.subtype_id.id == self.ref('website_blog.mt_blog_blog_published')), None)
         self.assertEqual(
             publish_message.notified_partner_ids,
             self.user_employee.partner_id | self.user_public.partner_id,
             'website_blog: peuple following a blog should be notified of a published post')
 
         # Armand posts a message -> becomes follower
-        test_blog_post.sudo().message_post(
+        self.test_blog_post.sudo().message_post(
             body='Armande BlogUser Commented',
             message_type='comment',
             author_id=self.user_employee.partner_id.id,
             subtype_xmlid='mail.mt_comment',
         )
         self.assertIn(
-            self.user_employee.partner_id, test_blog_post.message_partner_ids,
+            self.user_employee.partner_id, self.test_blog_post.message_partner_ids,
             'website_blog: people commenting a post should follow it afterwards')
+
+    @users('portal_user')
+    def test_blog_comment(self):
+        """Test comment on blog post with attachment."""
+        attachment = self.env['ir.attachment'].sudo().create({
+            'name': 'some_attachment.pdf',
+            'res_model': 'mail.compose.message',
+            'datas': 'test',
+            'type': 'binary',
+            'access_token': 'azerty',
+        })
+
+        with MockRequest(self.env):
+            PortalChatter().portal_chatter_post('blog.post', self.test_blog_post.id, 'Test message blog post', attachment_ids=str(attachment.id), attachment_tokens=attachment.access_token)
+
+        self.assertTrue(self.env['mail.message'].sudo().search([('model', '=', 'blog.post'), ('attachment_ids', 'in', attachment.ids)]))
+
+        second_attachment = self.env['ir.attachment'].sudo().create({
+            'name': 'some_attachment.pdf',
+            'res_model': 'mail.compose.message',
+            'datas': 'test',
+            'type': 'binary',
+            'access_token': 'azerty',
+        })
+
+        with self.assertRaises(Exception):
+            with MockRequest(self.env):
```

Comment:
Could combine them in one line, though it doesn't really matter.
```suggestion
        with self.assertRaises(Exception), MockRequest(self.env):
```
However the assertion on the exception is a bit weak as it is here, as any exception during code execution will be considered a success for this test, even though the only expected one is the actual access error related to the attachments.